src/Abstractions/Entities/TaskOrchestrationEntityFeature.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.DurableTask.Entities;
+
+/// <summary>
+/// Feature for interacting with durable entities from an orchestration.
+/// </summary>
+public abstract class TaskOrchestrationEntityFeature
+{
+    /// <summary>
+    /// Calls an operation on an entity and waits for it to complete. Does not wait for completion if
+    /// <see cref="CallEntityOptions.Signal"/> is set on <paramref name="options"/>.
+    /// </summary>
+    /// <typeparam name="TResult">The result of the entity operation.</typeparam>
+    /// <param name="id">The target entity.</param>
+    /// <param name="operationName">The name of the operation.</param>
+    /// <param name="input">The operation input.</param>
+    /// <param name="options">The call options.</param>
+    /// <returns>
+    /// The result of the entity operation, or default(<typeparamref name="TResult"/>) in the signal-only case.
+    /// </returns>
+    public abstract Task<TResult> CallEntityAsync<TResult>(
```

Comment:
Not sure I like the idea of using "CallEntityAsync" for both calls and signals. From a terminology perspective, we are not considering signals to be a special type of call, they are separate category of communication. Another issue which gets obscured here is that there are significant differences in the parameters and return values:
- there are no call options, only signal options. 
- signal never returns a TResult value.
- signal does not return a Task, it returns void, and is not async.

Reply:
Keeping implementations flexible is probably a good idea. 
If you think it fits better with the other patterns, you can keep it. Personally I would still change it, also just for discoverability (it's not obvious that to find the signal you have to pass an option to call).